src/Daemon/SentryWriter.cpp

```diff
@@ -133,35 +141,43 @@ void SentryWriter::initialize(Poco::Util::LayeredConfiguration & config)
     }
 }
 
-void SentryWriter::shutdown()
+SentryWriter::~SentryWriter()
 {
     if (initialized)
         sentry_shutdown();
 }
 
-void SentryWriter::onFault(int sig, const std::string & error_message, const StackTrace & stack_trace)
+void SentryWriter::onFault(int sig_or_error, const std::string & error_message, const FramePointers & frame_pointers, size_t offset, size_t size)
```

Comment:
I couldn't find definite answer in sentry docs about thread-safety. [They](https://docs.sentry.io/platforms/native/performance/instrumentation/custom-instrumentation/#thread-safety) mentioned that if something is not thread safe, then it is mentioned in the function's docs. `sentry_value_new_message_event` doesn't mention that, so my assumption is it is thread-safe. If I am right, then it is fine, if not, then we should make sure two logical errors happening at the same time won't cause any issues.

Previously we only reported crashes, so I am not sure if anybody thought about thread-safety, therefore I think we should make sure it is thread safe.

Reply:
Yes, it looks safe, global variables guarded with mutex (thought 100% it is hard to tell of course).